### DISCUSSION TITLE
sea-orm cannot to connect to a planetscale db

### DISCUSSION THREAD

**Discussion by lmnzx** [SeaQL/sea-orm#701] [Q&A] [answered]
Planeetscale is built on top of Vitress and MySQL driver is used in communicating with the database. I have used the mysql cargo package to test the connection and it works properly but when using the sea-orm-cli to generate entity it's giving an error

```shell
thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: ColumnDecode 
{ index: "0", source: "mismatched types; Rust type `alloc::string::String` (as SQL type `VARCHAR`) 
is not compatible with SQL type `VARCHAR`" }', 
/Users/UserName/.cargo/registry/src/github.com-1ecc6299db9ec823/sqlx-core-0.5.13/src/row.rs:73:37
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

How to fix this??

**Comment by vokup** [comment]
Hello, I also got the same error but I generate an entity by hand (similar to example) not by CLI. a connection to the `PlanetScale` is fine, but the error appeared when I try to query via `Entity::find().all(&db)`,. I also tried with column type `VARCHAR(21)`, `VARCHAR(255)`, `MEDIUMTEXT` and got the same error. I think it happened to all string-related column types. I also tried to execute simple query like `SELECT 1 + 1`  to test the connection inside and outside transaction and everything works fine. And if I remove the string-related column from the entity everything also works perfectly.

The error that appeared
```
thread 'main' panicked at 'called `Result::unwrap()` on an `Err` 
value: Query("error occurred while decoding column \"id\": mismatched types; 
Rust type `core::option::Option<alloc::string::String>` (as SQL type `VARCHAR`) 
is not compatible with SQL type `VARCHAR`")', src\main.rs:41:14
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

Entity Definition
```rs
#[derive(Clone, Debug, PartialEq, DeriveEntityModel)]
#[sea_orm(table_name = "dev_Dummy")]
pub struct Model {
    #[sea_orm(primary_key)]
    // I also tried
    // #[sea_orm(primary_key, column_type = "String(Some(21))")]
    pub id: String,
}
```

Thank you for your awesome library! .

**Comment by billy1624** [comment, marked as answer]
Hey @snxk & @vokup, thanks for raising this up. I found related discussion below
- https://github.com/planetscale/discussion/discussions/162

**Comment by tyt2y3** [comment]
However, since Planetscale is not an OSS project, it is unlikely we will support it officially without someone paying for it.
